sdks/java/core/src/main/java/org/apache/beam/sdk/util/GcsUtil.java

```diff
@@ -349,16 +350,45 @@ public boolean bucketExists(GcsPath path) throws IOException {
   }
 
   /**
+   * Returns the project number of the project which owns this bucket.
+   * If the bucket exists, it must be accessible otherwise the permissions
+   * exception will be propagated.
+   */
+  public long bucketOwner(GcsPath path) throws IOException {
+    return getBucket(
+        path,
+        BACKOFF_FACTORY.backoff(),
+        Sleeper.DEFAULT).getProjectNumber().longValue();
```

Comment:
`getBucket` will return `null` if the bucket is not accessible or does not exist - and we can't determine which one is the case from the returned `null`

This probably should be a propagated (potentially wrapped) exception from the internal `getBucket`, with `bucketExists` returning the appropriate value depending on the exception.